reference/client/get-pages.mdx

```diff
@@ -0,0 +1,366 @@
+---
+title: "getPages"
+description: "An instance method that provides data about the pages you've created within Makeswift"
+---
+
+## Options
+
+<ParamField query="limit" type="number" default={20}>
+  Number of pages to fetch. Must be between `1` and `100`.
+</ParamField>
+
+<ParamField query="after" type="string">
+  Starting `id` cursor of the last page entry that was fetched.
+</ParamField>
+
+<ParamField query="pathPrefix" type="string" default="/">
+  A filter to only retrieve pages with a pathname that begins with this value.
+</ParamField>
+
+<ParamField query="locale" type="string">
+  The site locale to fetch. 
+</ParamField>
+
+<ParamField query="sortBy" type="string" default="path"> 
+  The field to sort the result pages by. Can be any one of the following: `'title'`, `'path'`, `'description'`, `'createdAt'`, `'updatedAt'`.
+</ParamField>
+
+<ParamField query="sortDirection" type="string" default="asc">
+  The direction of how the result pages are sorted. Can be either `'asc'` or `'desc'`.
+</ParamField>
+
+<ParamField query="siteVersion" type="string" default={`Live`}>
+  Used to either retrieve the published version or working version of Makeswift
+  pages. Can either be `Live` or `Working` 
+</ParamField>
+
+<ParamField query="includeOffline" type="boolean" default={false}>
+  Whether the results should include pages that are offline in Makeswift.
+</ParamField>
+
+## Return type
+
+```ts
+type MakeswiftPage = {
+  id: string,
+  path: string,
+  title: string | null,
+  description: string | null,
+  canonicalUrl: string | null,
+  socialImageUrl: string | null,
+  sitemapPriority: number | null,
+  sitemapFrequency: string | null,
+  createdAt: string,
+  updatedAt: string,
+  publishedAt: string | null,
+  isOnline: boolean | null,
+  excludedFromSearch: boolean | null,
+  locale: string,
+  localizedVariants: {
+    locale: string,
+    path: string
+  }[]
+}
+
+// Final return type
+type MakeswiftGetPagesResult = {
+  data: MakeswiftPage[],
+  hasMore: boolean
+}
+```
+
+## Concepts
+
+### Pagination & Async Iterators
+
+Pagination will only retrieve a portion of the total pages per request. If you
+have a large number of pages, you may need to paginate through the results to
+retrieve all of them.
+
+To make this easier, we've implemented the return type of `getPages` as an
+extension of a `Promise` and an async iterator. This abstraction enables you to
+consume the pages in multiple ways, enabling you to easily retrieve all pages
+without worrying about the implementation details of pagination:
```

Comment:
A couple of things here:
- The returned value implements the `AsyncIterable` interface, not `AsyncIterator`
- "we've implemented the return type" sounds weird—what about "the returned value of `getPages` implements..."
- Maybe we should have a place in our docs for pagination that we can link from here where it explains how our paginated methods return a value that implement the `Thenable` and `AsyncIterable` interfaces in addition to providing some helpful utilities like `map`, `filter`, and `toArray`. Documenting this `PaginationResult` type seems like an important thing to do if we're gonna use it more and more. It's not something we have to do right now, but it came to mind.

Reply:
Addressed the first two bullets. As for the last, I agree - just like how the Stripe documents have a separate page just for pagination, I want there to be a separate page for us as well once we adopt this pattern across more bulk endpoints.